### DISCUSSION TITLE
MOD3.0 Design issue #2: How to capture the notion of “distribution”?

### DISCUSSION THREAD

**Discussion by jonquet** [sifrproject/MOD-Ontology#35] [General]
With the SemanticArtefact class being subcass of dcat:Dataset, we can:

### Proposition 1 : 
- SA specializes dcat:Resource only
- SADistribution is independant of dcat:Distribution but the idea of distribution object there.
- If one connects other object (e.g., schema:Dataset, void:Dataset) to dcat:Dataset then a SA can not benefit from their properties

<img width="524" alt="Capture d’écran 2023-04-10 à 15 00 53" src="https://user-images.githubusercontent.com/7379183/231007318-2662c155-0682-4526-b457-e08868dbb503.png">

### Proposition 2
- SA becomes a sub class of dcat:Dataset and thus SADistribution becomes a sub class of dcat:Distribution 
- The idea of keeping the distribution is till there but is a SA a dataset ?
- If we connect schema:Dataset to dcat:Dataset then a SA can still benefit from the properties
- We probably dont need the mod:distribution property anymore

<img width="423" alt="image" src="https://user-images.githubusercontent.com/7379183/231006875-a96d65fd-97af-4f5b-aa42-5ca3ba5a3f65.png">

### Proposition 3
- Get rid of SADistribution completely => but then it does not make sense to follow DCAT

**Comment by jonquet** [comment]
Properties available from dcat:distribution

<img width="234" alt="image" src="https://user-images.githubusercontent.com/7379183/231007670-51995aa8-753d-49a4-a16b-061040435e69.png">

**Comment by agbeltran** [comment]
Thanks @jonquet for summarising the 3 propositions.

I do think it is valuable to use DCAT's dichotomy between Dataset and Distribution for the specific case of SemanticArtefacts. 

So, if we define ```mod:SemanticArtefact``` as a ```dcat:Dataset``` as per #34, we can use ```dcat:Distribution``` without specialising it with a class ```mod:SemanticArtefactDistribution```. Basically, I am saying that we can go for proposition 3, but still use ```dcat:Distribution``` for the different representations of the semantic artefacts. Do we really need to define the class ```mod:SemanticArtefactDistribution```?

**Reply by jonquet** [reply]
I think yes we need to specialize the ```dcat:Distribution`` class in order to capture atributes that are specific to SA distribution and not necessarily common to every Distribution.

The diagram here illustrate all the propereties that are specific. 
<img width="459" alt="Capture d’écran 2023-07-05 à 10 54 07" src="https://github.com/FAIR-IMPACT/MOD/assets/7379183/b135eea5-f0df-49bd-a4f6-04752d84791f">

So this bring us back to Proposition 2

**Comment by agbeltran** [comment]
Following up on this design issue, we had a brief discussion on this in the last T4.2.3 meeting (29th September).

While some of the properties listed in the last diagram (Proposition 2) make sense to be in a ```mod:SemanticArtefactDistribution``` (assuming we extend ```dcat:Distribution```), we would need to make sure that all of them are at the level of distribution rather than at the level of dataset.

**Comment by jonquet** [comment]
Histroically discussed in MOD design. 
Discussed in FAIR-IMPACT T4.2.3 too on this day. 

?? The metrics properties could be positionned on the mod:SemanticArtefact ojbect ...but after debate, I still think its more "relevant" to have them at the distribution level. 
@agbeltran  @dgarijo

3 examples: 
- The distribution can be slightly different in number of axioms still it would be the "same ontology"
- The distribution can fo instance include only the classes that have French label, then the number of classes will be different (if not all classes have labels in French), still it would be the "same ontology"
- The distribution may use skos:prefLabel to name classes rather than rdfs:label, still it would be the "same ontology"

**Reply by dgarijo** [reply]
is there any property that would be affected by this? For instance, the number of classes and properties would be the same, and hence it makes sense to have them at the ontology level (semantic resource)

The number of annotations would differ in distributions per your example, I buy that argument. Or the property indicating the synonym property (although this is a use case where you are not describing the distribution itself, but the usage of the distribution...).

**Comment by agbeltran** [comment]
Summary of discussion (some of this derives from discussions in https://github.com/FAIR-IMPACT/MOD/discussions/34):
- MOD will continue the approach of DCAT and distinguish between SA and SADistribution
- SADistribution will extend dcat:Distribution
- mod:distribution is not needed, and dcat:distribution can be reused
- the properties listed for SADistribution will be analysed in case they need to move to SA, but overall it makes sense to consider properties for the abstract SA concept versus the realisation of a SA in a specific distribution
